format/index.md

```diff
@@ -26,64 +26,24 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
-application/vnd.api+json; supported-ext="bulk,jsonpatch"` indicates that the
-server supports both the "bulk" and "jsonpatch" extensions.
-
-If an extension is used to form a particular request or response document,
-then it **MUST** be specified by including its name in the `ext` media type
-parameter with the `Content-Type` header. The value of the `ext` media type
-parameter **MUST** be formatted as a comma-separated (U+002C COMMA, ",")
-list of extension names and **MUST** be limited to a subset of the
-extensions supported by the server, which are listed in `supported-ext`
-of every response.
-
-For example: a response that includes the header `Content-Type:
-application/vnd.api+json; ext="ext1,ext2"; supported-ext="ext1,ext2,ext3"`
-indicates that the response document is formatted according to the
-extensions "ext1" and "ext2". Another example: a request that includes
-the header `Content-Type: application/vnd.api+json; ext="ext1,ext2"`
-indicates that the request document is formatted according to the
-extensions "ext1" and "ext2".
-
-Clients **MAY** request a particular media type extension by including its
-name in the `ext` media type parameter with the `Accept` header. Servers
-that do not support a requested extension or combination of extensions
-**MUST** return a `406 Not Acceptable` status code.
-
-If the media type in the `Accept` header is supported by a server but the
-media type in the `Content-Type` header is unsupported, the server
-**MUST** return a `415 Unsupported Media Type` status code.
-
-Servers **MUST NOT** provide extended functionality that is incompatible
-with the base specification to clients that do not request the extension in
-the `ext` parameter of the `Content-Type` or the `Accept` header.
-
-> Note: Since extensions can contradict one another or have interactions
-that can be resolved in many equally plausible ways, it is the
-responsibility of the server to decide which extensions are compatible, and
-it is the responsibility of the designer of each implementation of this
-specification to describe extension interoperability rules which are
-applicable to that implementation.
-
-When the value of the `ext` or `supported-ext` media type parameter contains
-more than one extension name, the value **MUST** be surrounded with quotation
-marks (U+0022 QUOTATION MARK, """), in accordance with the HTTP specification.
+## Media Type Negotiation <a href="#media-type-negotiation" id="media-type-negotiation" class="headerlink"></a>
 
+Clients **MUST** send all JSON API data in request documents with
+the header `Content-Type: application/vnd.api+json`.
+
+Clients **MUST** ignore all parameters for the `application/vnd.api+json`
+media type received in the `Content-Type` header of response documents.
+
+Servers **MUST** send all JSON API data in response documents with
+the header `Content-Type: application/vnd.api+json`.
+
+Servers **MUST** return a `406 Not Acceptable` status code if the
```

Comment:
I'd rewrite this fourth paragraph to be something like:

> Methods for extending this specification will be added at a later date. Servers that implement this specification before those extension mechanisms are defined, or that don't implement those mechanisms once they are defined, **MUST** respond with `400 Bad Request` to any request that includes a media type parameter in the `Content-Type` header. These servers also **MUST** respond with a `400 Bad Request` for any request that, if it lists the JSON API media type in its `Accept` header, only lists parameterized forms of it.

The key thing going on here is that we're making requests with media type parameters explicit `400` errors, rather than letting the server ignore those parameters or respond with `406` or `415`. Requiring an explicit error will make it easier, when we add extensions in the future, for clients to distinguish between _extensions not being supported at all_ vs. _a particular extension not being supported_. 

The other thing I tried to do was clarify that the `Accept` header can take multiple values, e.g. `Accept: application/vnd.api+json;ext=bulk, application/vnd.api+json` and that the header is only problematic if **every** JSON-API option it lists contains parameters. If only some of the options contain parameters (as in the prior example), then the requests should succeed—and, in fact, future clients may want to format their `Accept` header like this to work with older servers.

(Additionally, we could add that, in the 400 response, the server **SHOULD** (must?) return an error object with a particular name and/or code.)